addons/account/tests/test_account_move_entry.py

```diff
@@ -677,3 +677,32 @@ def test_misc_prevent_edit_tax_on_posted_moves(self):
         # You can remove journal items if the related journal entry is draft.
         self.test_move.button_draft()
         self.assertTrue(self.test_move.line_ids.unlink())
+
+    def test_draft_name_on_move_to_period(self):
+        # When a draft entry is added to an empty period, it should get a name.
+        # When a draft entry with a name is moved to a period already having entries, its name should be reset to '/'.
+        new_move = self.env['account.move'].create({
+            'move_type': 'entry',
+            'date': fields.Date.from_string('2016-02-01'),
```

Comment:
`from_string` is deprecated
```suggestion
            'date': fields.Date.to_date('2016-02-01'),
```